pkg/cleanup/containers.go

```diff
@@ -23,30 +24,19 @@ func (c *containers) Name() string {
 	return "containers steps"
 }
 
-// NeedsToRun checks if custom CRI is used, otherwise checks if containerd is present on the host
-func (c *containers) NeedsToRun() bool {
-	if c.isCustomCriUsed() {
-		return true
-	}
-	if _, err := os.Stat(c.Config.containerd.binPath); err != nil {
-		logrus.Debugf("could not find containerd binary at %v errored with: %v", c.Config.containerd.binPath, err)
-		return false
-	}
-	return true
-}
-
 // Run removes all the pods and mounts and stops containers afterwards
 // Run starts containerd if custom CRI is not configured
 func (c *containers) Run() error {
 	if !c.isCustomCriUsed() {
 		if err := c.startContainerd(); err != nil {
-			logrus.Debugf("error starting containerd: %v", err)
-			return err
+			if os.IsNotExist(err) {
+				logrus.Debugf("containerd binary not found. Skipping container cleanup")
+				return nil
+			}
+			return fmt.Errorf("failed to start containerd: %v", err)
```

Comment:
I think these `return fmt.Errorf("something: %v", err")` are supposed to use `%w` instead of `%v` for some fancy error unwrapping stuff we don't probably use.

Reply:
fixed.